ampersand-state.js

```diff
@@ -394,8 +402,8 @@ assign(Base.prototype, Events, {
             derived: false
         }, options || {});
         var res = {};
-        var val, item, def;
-        for (item in this._definition) {
+        var val, def;
+        for (var item in this._definition) {
```

Comment:
Defensive `var` - in general this is good practice, just in case a refactoring moves this loop into another scope which [infamously causes deopt](https://github.com/petkaantonov/bluebird/wiki/Optimization-killers#5-for-in).